### PR TITLE
octoprint: add override for celery and sentry-sdk, remove constraint of sarge

### LINKED ISSUE #151456: Octoprint not building on unstable/master
### Describe the bug

Octoprint can't build on unstable due to a conflict of the required version for `click`.
octoprint [has an override for `click-7.1.2`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/octoprint/default.nix#L30), while the dependency `sentry-sdk` requires (due to `celery-5.2.1`) `click<9.0,>=8.0`. 

### Steps To Reproduce

```console
nix-build '<nixpkgs>' -A octoprint
```

### Expected behavior
Octoprint builds successfully

### Additional context

I ran a `git bisect` between 54c0954df4cd6f6feb6d2bb3a57c2e950ebd4bb2 and 3d20b6a4712399d900cc4b7923be05dd896f5dc3 (known good) , the first bad commit was f56b127d07ef5632aae1b721d0e2db2f9682375d

I suspect that it's necessary to pin the additional dependencies `sentry-sdk` (and maybe some more), too, but I'm not that experienced with packaging python packages.

### Notify maintainers

@fabaff made the first bad commit

Octoprint maintainers: @abbradar @gebner @WhittlesJr   

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.81, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"home-manager, nixos-22.05pre339530.8021ed20904, nixos-hardware, nixos-mailserver, nur"`
 - channels(richard): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: pkgs.octoprint
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Fix build

Fixes #151456

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 151577` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>gns3-server</li>
    <li>home-assistant</li>
  </ul>
</details>
<details>
  <summary>32 packages built:</summary>
  <ul>
    <li>alerta-server</li>
    <li>cura</li>
    <li>gns3-gui</li>
    <li>moodle-dl</li>
    <li>moz-phab</li>
    <li>octoprint</li>
    <li>python38Packages.celery</li>
    <li>python38Packages.django-raster</li>
    <li>python38Packages.django-rq</li>
    <li>python38Packages.djmail</li>
    <li>python38Packages.flower</li>
    <li>python38Packages.sarge</li>
    <li>python38Packages.sentry-sdk</li>
    <li>python39Packages.celery</li>
    <li>python39Packages.django-raster</li>
    <li>python39Packages.django-rq</li>
    <li>python39Packages.djmail</li>
    <li>python39Packages.flower</li>
    <li>python39Packages.sarge</li>
    <li>python39Packages.sentry-sdk</li>
    <li>sourcehut.buildsrht</li>
    <li>sourcehut.coresrht</li>
    <li>sourcehut.dispatchsrht</li>
    <li>sourcehut.gitsrht</li>
    <li>sourcehut.hgsrht</li>
    <li>sourcehut.hubsrht</li>
    <li>sourcehut.listssrht</li>
    <li>sourcehut.mansrht</li>
    <li>sourcehut.metasrht</li>
    <li>sourcehut.pastesrht</li>
    <li>sourcehut.todosrht</li>
    <li>tribler</li>
  </ul>
</details>

File: pkgs/applications/misc/octoprint/default.nix [hunk 1]
```diff
@@ -122,6 +122,45 @@ let
...
+        # Octoprint pulls in celery indirectly but has no support for the up-to-date releases
+        (
+          self: super: {
+            celery = super.celery.overrideAttrs (oldAttrs: rec {
+              version = "5.0.0";
+              src = oldAttrs.src.override {
+                inherit version;
+                sha256 = "sha256-MTkw/d3nA9jjcCmjBL+RQpzRGu72PFfebayp2Vjh8lU=";
+              };
+              doCheck = false;
```
Comment (risicle) [inline pkgs/applications/misc/octoprint/default.nix:134]: Add `pythonImportsCheck` as substitute?

Comment (risicle) [reply pkgs/applications/misc/octoprint/default.nix:134]: Oh you did.

**Comment by risicle** [comment]
`nixpkgs-review` reveals no new failures, macos 10.15, nixos x86_64 & aarch64

**Pull request merged**